generated/nidcpower/session.py

```diff
@@ -4151,6 +4151,37 @@ def export_attribute_configuration_file(self, file_path):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    @ivi_synchronized
+    def create_advanced_sequence(self, sequence_name, sequence, set_as_active_sequence=True):
+        '''create_advanced_sequence
+
+        Test.
+
+        Note: Test.
+
+        Args:
+            sequence_name (str): Name of sequence.
+
+            sequence (list of int): Sequence.
+
+            set_as_active_sequence (bool): Specifies that this current sequence is active.
+
+        '''
+        # First we need to get all possible properties we might be setting
+        attribute_ids_used = set()
+        for s in sequence:
+            for key in s:
+                attribute_ids_used.add(Session.__base__.__dict__[key]._attribute_id)
```

Comment:
While this is nice and concise, one disadvantage is that if the client passes a string that's not the name of an attribute, the error will be pretty bad. If the string matches another non-attribute member it will be one thing. If the string doesn't match any members, it will be another thing. This would be a common thing with typos. Do you think the error raised is good enough?

Reply:
Added error checking for both existence and type of the key